.github/workflows/_build.yml

```diff
@@ -175,10 +174,76 @@ jobs:
       - name: Run Tests
         run: ./scripts/ci ${{ matrix.schemes }} test
 
+  Screenshots:
+    name: Screenshots tests
+    runs-on: macos-latest
+    needs: [Setup]
+    permissions:
+      pull-requests: write
+      contents: write
+    timeout-minutes: 20
+    continue-on-error: true
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Bundler install Gems
+        uses: actions/cache@v4
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ env.ImageVersion }}-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-${{ env.ImageVersion }}
+
+      - name: Cache Pods
+        uses: actions/cache@v4
+        with:
+          path: Example/Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
+
+      - name: Bundle Install
+        run: bundle install --jobs 4 --retry 3
+
+      - name: Pod Install
+        run: bundle exec pod install
+        working-directory: Example
+
+      - name: Select Xcode 14
+        run: sudo xcode-select --switch /Applications/Xcode_14.2.app
+      
+      - name: Screenshot Tests
+        id: screenshotTests
+        run: |
+          ./scripts/record_screenshot_tests.sh
+
+          changedFiles=`git status --porcelain` && echo "CHANGED_FILES=${changedFiles//$'\n'/'%0A'}" >> $GITHUB_OUTPUT
+        continue-on-error: true
+
+      - name: Check snapshot changes
+        id: checkSnapshotChanges
+        if: ${{ github.event_name != 'pull_request' }}
+        run: ./scripts/lint
+
+      - name: Commit changes
+        if: ${{ github.event_name == 'pull_request' && steps.screenshotTests.outputs.CHANGED_FILES != '' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git remote set-url origin https://x-access-token:$GITHUB_TOKEN@github.com/${{ github.repository }}
+          git fetch origin $GITHUB_HEAD_REF
+          git checkout $GITHUB_HEAD_REF
+          git add */**/__Snapshots__/*.png
+          git diff-index --quiet HEAD || git commit -m "Updated snapshots"
+          git push
+
   UploadArtifacts:
     name: Upload Artifacts
     runs-on: macos-latest
-    needs: [TestPods]
+    needs: [TestPods, Screenshots]
```

Comment:
so you need to upload these? wouldn't they be part of the pr already? (or it's in case they fail probably right?)

Reply:
Just want to create a 'flow' where upload artifacts is the last step.